corehq/apps/accounting/payment_handlers.py

```diff
@@ -400,24 +400,27 @@ def _pay_invoice(self, invoice):
             return
 
         try:
-            with transaction.atomic():
-                payment_record = PaymentRecord.create_record(payment_method, 'temp_transaction_id', amount)
-                invoice.pay_invoice(payment_record)
-                invoice.subscription.account.last_payment_method = LastPayment.CC_AUTO
-                invoice.account.save()
-                transaction_id = payment_method.create_charge(
-                    autopay_card,
-                    amount_in_dollars=amount,
-                    description='Auto-payment for Invoice %s' % invoice.invoice_number,
-                )
+            log_accounting_info("[Autopay] Attempt to charge autopay invoice {} through Stripe".format(invoice.id))
+            transaction_id = payment_method.create_charge(
+                autopay_card,
+                amount_in_dollars=amount,
+                description='Auto-payment for Invoice %s' % invoice.invoice_number,
+                idempotency_key=invoice.id
```

Comment:
As we discussed over voice, it might make sense to have this be a hash of `invoice.id` and `amount` in case we ever made some colossal mistake and somehow started sending duplicate invoice numbers.

Reply:
Sorry It's been a week and I forgot this. Is a simple combination of invoice number and amount enough? Or does it have to be a hash? I just recalled Biyeun mentioned using invoice number other than `invoice.id`, because India's database can have the same invoice id, but the invoice number will be different.

Reply:
> Is a simple combination of invoice number and amount enough?

Do mean something like `repr((invoice.id, amount))`? I suppose that would be fine if the `idempotency_key` allows characters such as spaces. And I suppose the length of that tuple repr would never exceed the maximum length supported by `idempotency_key`?

Reply:
I'm actually thinking of something like this `idempotency_key = f"{self.invoice_number()}_{amount}"`
Idempotency keys can be up to 255 characters long.